src/services/item/itemServiceManager.ts

```diff
@@ -0,0 +1,35 @@
+import { DataSource } from 'typeorm';
+
+import { ItemType } from '@graasp/sdk';
+
+import { FolderService } from './folderService';
+import { AppService } from './plugins/app/service';
+import { H5PService } from './plugins/html/h5p/service';
+import { ItemService } from './service';
+
+export class ItemServiceManager {
+  // get from injection
+  private db: DataSource;
+  private h5pService: H5PService;
+  private appService: AppService;
+  private folderService: FolderService;
+
+  static async getServiceForTypeFromId(repositories, id: string): Promise<ItemService> {
+    // TODO: check permission
+
+    // not really cool
+    const item = await repositories.itemRepository.get(id);
+
+    // question: would be best to get the service given the item
+    // like item.getService() but not sure it's feasible
```

Comment:
I'm wondering if it is the responsibility of the item to know which service to use, I'm not sure.